File: python/examples/quickstart.py
from roteiro import Pipeline, RoteiroClient


def main() -> None:
    client = RoteiroClient(
        base_url="http://localhost:8080",
        project_id=42,
    )

    health = client.health()
    bodies = client.list_bodies()
    result = Pipeline(client).buffer(100).simplify(10).execute("roads")

    print(
        {
            "health": health.status,
            "body_count": len(bodies.get("bodies", [])) if isinstance(bodies, dict) else None,
            "result": result,
        }
    )


if __name__ == "__main__":
    main()

File: typescript/examples/quickstart.ts
import { Pipeline, RoteiroClient } from '../src/index';

async function main(): Promise<void> {
  const client = new RoteiroClient({
    baseUrl: 'http://localhost:8080',
    projectId: 42,
  });

  const health = await client.health();
  const bodies = await client.listBodies();

  const result = await new Pipeline(client)
    .buffer(100)
    .simplify(10)
    .execute('roads');

  console.log({ health, bodyCount: Array.isArray((bodies as { bodies?: unknown[] }).bodies) ? (bodies as { bodies: unknown[] }).bodies.length : undefined, result });
}

main().catch((error) => {
  console.error(error);
  process.exitCode = 1;
});
